core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/buffer.go

```diff
@@ -62,6 +118,22 @@ func (b fetchedBlock) Clone() fetchedBlock {
 	}
 }
 
+// Sort by log identifiers, shuffled using a pseduorandom souce that is shared across all nodes
+// for a given block.
+func (b *fetchedBlock) Sort() {
+	randSeed := random.GetRandomKeySource(nil, uint64(b.blockNumber))
+
+	shuffledLogIDs := make(map[string]string, len(b.logs))
+	for _, log := range b.logs {
+		logID := log.getLogID()
+		shuffledLogIDs[logID] = random.ShuffleString(logID, randSeed)
```

Comment:
nit. i think this will always be constant for a given log, so we can also make it a part of logID itself